medusa/post_processor.py

```diff
@@ -148,85 +147,54 @@ def list_associated_files(self, file_path, base_name_only=False,
 
-            # supported subtitle languages codes
-            language_extensions = tuple('.' + c for c in language_converters['opensubtitles'].codes)
+        # loop through all the files in the folder, and check if they are the same name
+        # even when the cases don't match
+        filelist = []
+        for found_file in file_list:
 
-            # loop through all the files in the folder, and check if they are the same name
-            # even when the cases don't match
-            for filefound in checklist:
+            file_name = os.path.basename(found_file).lower()
 
-                file_name = filefound.rpartition('.')[0]
-                file_extension = filefound.rpartition('.')[2]
-                is_subtitle = None
+            if file_name[-3:] in subtitle_extensions:
+                code = file_name.rsplit('.', 2)[1].replace('_', '-')
+                language = from_code(code, unknown='') or from_ietf_code(code, unknown='und')
+                if language:
+                    filelist.append(found_file)
 
-                if file_extension in subtitle_extensions:
-                    is_subtitle = True
-
-                if not base_name_only:
-                    new_file_name = file_name + '.'
-                    sub_file_name = file_name.rpartition('.')[0] + '.'
-                else:
-                    new_file_name = file_name
-                    sub_file_name = file_name.rpartition('.')[0]
-
-                if is_subtitle and sub_file_name.lower() == base_name.lower().replace('[[]', '[').replace('[]]', ']'):
-                    extension_len = len(filefound.rsplit('.', 2)[1])
-                    if file_name.lower().endswith(language_extensions) and (extension_len in [2, 3]):
-                        filelist.append(filefound)
-                    elif file_name.lower().endswith('pt-br') and extension_len == 5:
-                        filelist.append(filefound)
-                # if there's no difference in the filename add it to the filelist
-                elif new_file_name.lower() == base_name.lower().replace('[[]', '[').replace('[]]', ']'):
-                    filelist.append(filefound)
+            # if there's no difference in the filename add it to the filelist
+            elif file_name.startswith(processed_file_name):
+                filelist.append(found_file)
 
+        file_path_list = []
+        extensions_to_delete = []
         for associated_file_path in filelist:
             # Exclude the video file we are post-processing
             if associated_file_path == file_path:
                 continue
 
-            # Exlude non-subtitle files with the 'only subtitles' option (not implemented yet)
-            # if subtitles_only and not associated_file_path[-3:] in subtitle_extensions:
-            #     continue
+            # Exlude non-subtitle files with the 'only subtitles' option
+            if subtitles_only and not associated_file_path[-3:] in subtitle_extensions:
```

Comment:
Same as before. Probably we should have a method: is_subtitle, like the one: isMediaFile